src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -8701,6 +8706,10 @@ def astype(
                 labels, include_index=False
             )
         )
+        data_column_snowpark_pandas_types = [
```

Comment:
let's make it more general (thinking about future types like interval)
we can do like this:
```
data_column_snowpark_pandas_types = [SnowparkPandasType.get_snowpark_pandas_type_for_pandas_type(t) for t in col_dtypes_map.values()]
```

Reply:
Need the instantiated version of the type so did this:
```
        data_column_snowpark_pandas_types = []
        for t in col_dtypes_map.values():
            snowpark_pandas_type = SnowparkPandasType.get_snowpark_pandas_type_for_pandas_type(t)
            data_column_snowpark_pandas_types.append(snowpark_pandas_type() if snowpark_pandas_type else None)
```